From 00b3f8968080eb269f205d188a7fe418ddd056dd Mon Sep 17 00:00:00 2001
From: Danny Kukawka <danny.kukawka@web.de>
Date: Tue, 20 Apr 2010 16:01:47 +0200
Subject: [PATCH 33/33] add a generic match for HP machines to fix brightness handling

Added a generic match for HP machines to fix brightness handling by
setting laptop_panel.brightness_in_hardware=false on all HP laptops
with a bios release date >= 2009. (bnc#587057)
---
 .../10freedesktop/10-laptop-panel-hardware.fdi     |   11 +++++++++--
 1 files changed, 9 insertions(+), 2 deletions(-)

diff --git a/fdi/information/10freedesktop/10-laptop-panel-hardware.fdi b/fdi/information/10freedesktop/10-laptop-panel-hardware.fdi
index 1b79e1e..6685ceb 100644
--- a/fdi/information/10freedesktop/10-laptop-panel-hardware.fdi
+++ b/fdi/information/10freedesktop/10-laptop-panel-hardware.fdi
@@ -37,8 +37,15 @@
       </match>
 
       <match key="/org/freedesktop/Hal/devices/computer:system.hardware.vendor" string="Hewlett-Packard">
-        <match key="/org/freedesktop/Hal/devices/computer:system.hardware.product" contains_outof="4310s;4410s;4411s;4415s;4416s;4510s;4515s;4710s;Compaq 5X5;Compaq 515;Compaq 516;Compaq 615;HP Mini 5102;EliteBook 8440w;EliteBook 8740w;EliteBook 8540p">
-          <merge key="laptop_panel.brightness_in_hardware" type="bool">false</merge>
+        <match key="/org/freedesktop/Hal/devices/computer:system.formfactor" string="laptop">
+          <match key="/org/freedesktop/Hal/devices/computer:system.hardware.product" contains_outof="4310s;4410s;4411s;4415s;4416s;4510s;4515s;4710s;Compaq 5X5;Compaq 515;Compaq 516;Compaq 615;HP Mini 5102">
+            <merge key="laptop_panel.brightness_in_hardware" type="bool">false</merge>
+          </match>
+          <match key="laptop_panel.brightness_in_hardware" bool="true">
+            <match key="/org/freedesktop/Hal/devices/computer:system.firmware.release_date" contains_outof="2009;2010;2011;2012;2013;2014;2015">
+              <merge key="laptop_panel.brightness_in_hardware" type="bool">false</merge>
+            </match>
+          </match>
         </match>
       </match>
 
-- 
1.7.1

